pymtl3/passes/rtlir/behavioral/BehavioralRTLIRTypeCheckL1Pass.py

```diff
@@ -150,7 +149,7 @@ def visit_Assign( s, node ):
     node.Type = None
 
   def visit_FreeVar( s, node ):
-    if node.name not in s.freevars.keys():
+    if node.name not in list(s.freevars.keys()):
```

Comment:
To check membership of keys in a dictionary, `if node.name not in s.freevars:` is both idiomatic and O(1) instead of O(n) - you do hash-lookup instead of constructing and traversing the list.